From cdd135c051ecfb6df81071643d961351e5a444e5 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Thu, 16 Jul 2020 16:05:08 +0800
Subject: [PATCH] arm64: dts: rockchip: update rk3399 and rk3399pro linux
 boards

1) remove fiq_debugger, exist on rk3399-linux.dtsi
2) update the related vpu node
3) update edp node

Signed-off-by: Caesar Wang <wxt@rock-chips.com>
Change-Id: I64f563bd75eb91d252681e6075b52d1715c7b3d4
---
 .../dts/rockchip/rk3399-firefly-android.dts   |  8 --
 .../dts/rockchip/rk3399-firefly-linux.dts     | 19 -----
 .../arm64/boot/dts/rockchip/rk3399-linux.dtsi | 10 +++
 ...ire-excavator-linux-for-rk1808-cascade.dts | 12 ---
 .../rk3399-sapphire-excavator-linux.dts       | 84 +++++++++++--------
 .../dts/rockchip/rk3399-videostrong-linux.dts |  6 --
 .../dts/rockchip/rk3399pro-evb-v10-linux.dts  | 12 ---
 .../dts/rockchip/rk3399pro-evb-v11-linux.dts  | 12 ---
 8 files changed, 60 insertions(+), 103 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
index 9a992991a1a9..cf25df4b77f1 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-android.dts
@@ -935,10 +935,6 @@
 	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
-&rkvdec {
-	status = "okay";
-};
-
 &rockchip_suspend {
 	rockchip,power-ctrl =
 		<&gpio1 18 GPIO_ACTIVE_LOW>,
@@ -1101,7 +1097,3 @@
 &vopl_mmu {
 	status = "okay";
 };
-
-&vpu {
-	status = "okay";
-};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
index 53725c338fa7..9efefdf7c7ef 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-firefly-linux.dts
@@ -121,17 +121,6 @@
 		};
 	};
 
-	fiq_debugger: fiq-debugger {
-		compatible = "rockchip,fiq-debugger";
-		rockchip,serial-id = <2>;
-		rockchip,signal-irq = <182>;
-		rockchip,wake-irq = <0>;
-		rockchip,irq-mode-enable = <1>;  /* If enable uart uses irq instead of fiq */
-		rockchip,baudrate = <1500000>;  /* Only 115200 and 1500000 */
-		pinctrl-names = "default";
-		pinctrl-0 = <&uart2c_xfer>;
-	};
-
 	gpio-keys {
 		compatible = "gpio-keys";
 		#address-cells = <1>;
@@ -905,10 +894,6 @@
 	pinctrl-0 = <&pwm2_pin_pull_down>;
 };
 
-&rkvdec {
-	status = "okay";
-};
-
 &rockchip_suspend {
 	rockchip,power-ctrl =
 		<&gpio1 18 GPIO_ACTIVE_LOW>,
@@ -1084,7 +1069,3 @@
 &vopl_mmu {
 	status = "okay";
 };
-
-&vpu {
-	status = "okay";
-};
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
index 54479ae63e82..952d711f6e07 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-linux.dtsi
@@ -210,6 +210,16 @@
 	status = "okay";
 };
 
+&rkvdec {
+	status = "okay";
+	/* 0 means ion, 1 means drm */
+	//allocator = <0>;
+};
+
+&rkvdec_mmu {
+	status = "okay";
+};
+
 &vdpu {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts
index 8df794fe7a22..7e93b9e05c88 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts
@@ -216,18 +216,6 @@
 	vref-supply = <&vccadc_ref>;
 };
 
-&vpu {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
-&rkvdec {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
 &backlight {
 	status = "okay";
 	enable-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
index 0be01405668a..892fef952753 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
@@ -50,28 +50,49 @@
 	model = "Rockchip RK3399 Excavator Board (Linux Opensource)";
 	compatible = "rockchip,rk3399-excavator-linux", "rockchip,rk3399";
 
-	fiq_debugger: fiq-debugger {
-		compatible = "rockchip,fiq-debugger";
-		rockchip,serial-id = <2>;
-		rockchip,signal-irq = <182>;
-		rockchip,wake-irq = <0>;
-		rockchip,irq-mode-enable = <1>;  /* If enable uart uses irq instead of fiq */
-		rockchip,baudrate = <1500000>;  /* Only 115200 and 1500000 */
-		pinctrl-names = "default";
-		pinctrl-0 = <&uart2c_xfer>;
+	vcc_lcd: vcc-lcd {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc_lcd";
+		gpio = <&gpio4 30 GPIO_ACTIVE_HIGH>;
+		startup-delay-us = <20000>;
+		enable-active-high;
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-boot-on;
+		vin-supply = <&vcc5v0_sys>;
 	};
 
-	edp_panel: edp-panel {
-		compatible = "lg,lp079qx1-sp0v", "panel-simple";
+	panel: panel {
+		compatible = "simple-panel";
 		backlight = <&backlight>;
-		power-supply = <&vcc3v3_s0>;
-		enable-gpios = <&gpio4 30 GPIO_ACTIVE_HIGH>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&lcd_panel_reset>;
+		power-supply = <&vcc_lcd>;
+		enable-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+		prepare-delay-ms = <20>;
+		enable-delay-ms = <20>;
+
+		display-timings {
+			native-mode = <&timing0>;
+
+			timing0: timing0 {
+				clock-frequency = <200000000>;
+				hactive = <1536>;
+				vactive = <2048>;
+				hfront-porch = <12>;
+				hsync-len = <16>;
+				hback-porch = <48>;
+				vfront-porch = <8>;
+				vsync-len = <4>;
+				vback-porch = <8>;
+				hsync-active = <0>;
+				vsync-active = <0>;
+				de-active = <0>;
+				pixelclk-active = <0>;
+			};
+		};
 
 		ports {
-			panel_in_edp: endpoint {
-				remote-endpoint = <&edp_out_panel>;
+			panel_in: endpoint {
+				remote-endpoint = <&edp_out>;
 			};
 		};
 	};
@@ -253,21 +274,15 @@
 	vref-supply = <&vccadc_ref>;
 };
 
-&vpu {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
-&rkvdec {
+&backlight {
 	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
+	enable-gpios = <&gpio4 29 GPIO_ACTIVE_HIGH>;
 };
 
-&backlight {
+&cdn_dp {
 	status = "okay";
-	enable-gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+	extcon = <&fusb0>;
+	phys = <&tcphy0_dp>;
 };
 
 &display_subsystem {
@@ -283,14 +298,11 @@
 	force-hpd;
 
 	ports {
-		edp_out: port@1 {
+		port@1 {
 			reg = <1>;
-			#address-cells = <1>;
-			#size-cells = <0>;
 
-			edp_out_panel: endpoint@0 {
-				reg = <0>;
-				remote-endpoint = <&panel_in_edp>;
+			edp_out: endpoint {
+				remote-endpoint = <&panel_in>;
 			};
 		};
 	};
@@ -384,6 +396,8 @@
 
 &vopb {
 	status = "okay";
+	assigned-clocks = <&cru DCLK_VOP0_DIV>;
+	assigned-clock-parents = <&cru PLL_CPLL>;
 };
 
 &vopb_mmu {
@@ -392,6 +406,8 @@
 
 &vopl {
 	status = "okay";
+	assigned-clocks = <&cru DCLK_VOP1_DIV>;
+	assigned-clock-parents = <&cru PLL_VPLL>;
 };
 
 &vopl_mmu {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts
index 0d0bad8561cf..d4218d279a79 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-videostrong-linux.dts
@@ -254,12 +254,6 @@
 	status = "okay";
 };
 
-&vpu {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
 &vopb {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
index 402114f1a4fe..101f724c5116 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
@@ -896,12 +896,6 @@
 	};
 };
 
-&rkvdec {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
 &rockchip_suspend {
 	status = "okay";
 	rockchip,sleep-debug-en = <1>;
@@ -1046,12 +1040,6 @@
 	status = "okay";
 };
 
-&vpu {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
 &pinctrl {
 	pinctrl-names = "default";
 	pinctrl-0 = <&npu_ref_clk>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
index c282e80e3755..eda40626500b 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
@@ -902,12 +902,6 @@
 	};
 };
 
-&rkvdec {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
 &rockchip_suspend {
 	status = "okay";
 	rockchip,sleep-debug-en = <1>;
@@ -1052,12 +1046,6 @@
 	status = "okay";
 };
 
-&vpu {
-	status = "okay";
-	/* 0 means ion, 1 means drm */
-	//allocator = <0>;
-};
-
 &pinctrl {
 	pinctrl-names = "default";
 	pinctrl-0 = <&npu_ref_clk>;
-- 
2.35.3

